### PR TITLE
Expiry Tests

### PR DESCRIPTION
**Pull request by OlafSzmidt**
Implements #320.

### CODE REVIEW

**Comment by OlafSzmidt** [comment]
Change list() [0] as you can't make sure the order of the set to list.

File: aimmo-game/tests/functional/test_effect_expiry.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        self.cell.pickup = pickup_created
+        self.assertEqual(self.avatar.resistance, 0)
+
+        # Avatar moves EAST to (1,0) where pickup is located, then repeats it 5 times.
+        for i in range(6):
+            self.game.turn_manager._run_single_turn()
+
+        self.assertTrue(isinstance(list(self.avatar.effects)[0], pickup_created.EFFECT))
+        self.assertEqual(list(self.avatar.effects)[0]._time_remaining, 5)
+        self.assertEqual(self.avatar.resistance, 1000)
```
Comment (CelineBoudier) [inline aimmo-game/tests/functional/test_effect_expiry.py]: where does the 1000 come from?

File: aimmo-game/tests/functional/test_effect_expiry.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+        self.assertEqual(self.avatar.resistance, 0)
+
+        # Avatar moves EAST to (1,0) where pickup one is located.
+        self.game.turn_manager._run_single_turn()
+
+        self.assertTrue(isinstance(list(self.avatar.effects)[0], pickup_created_one.EFFECT))
+        self.assertEqual(len(self.avatar.effects), 1)
+        self.assertEqual(list(self.avatar.effects)[0]._time_remaining, 10)
+        self.assertEqual(self.avatar.resistance, 1000)
```
Comment (CelineBoudier) [inline aimmo-game/tests/functional/test_effect_expiry.py]: same

File: aimmo-game/tests/functional/test_effect_expiry.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        self.assertEqual(list(self.avatar.effects)[1]._time_remaining, 10)
+        self.assertEqual(self.avatar.resistance, 2000)
+
+        # Eight turns later, we expect the first effect to expire.
+        for i in range(8):
+            self.game.turn_manager._run_single_turn()
+
+        self.assertEqual(len(self.avatar.effects), 1)
+        self.assertEqual(list(self.avatar.effects)[0]._time_remaining, 2)
+        self.assertEqual(self.avatar.resistance, 1000)
```
Comment (CelineBoudier) [inline aimmo-game/tests/functional/test_effect_expiry.py]: same

File: aimmo-game/tests/functional/test_effect_expiry.py (hunk 1 above)
Comment (OlafSzmidt) [reply aimmo-game/tests/functional/test_effect_expiry.py]: That's the resistance effect that the InvulnerabilityPickupEffect causes the avatar to have.

File: aimmo-game/tests/functional/test_effect_expiry.py (hunk 3 above)
Comment (OlafSzmidt) [reply aimmo-game/tests/functional/test_effect_expiry.py]: That's the resistance effect that the InvulnerabilityPickupEffect causes the avatar to have.

File: aimmo-game/tests/functional/test_effect_expiry.py (hunk 2 above)
Comment (OlafSzmidt) [reply aimmo-game/tests/functional/test_effect_expiry.py]: That's the resistance effect that the InvulnerabilityPickupEffect causes the avatar to have.

File: aimmo-game/tests/functional/test_effect_expiry.py (hunk 1 above)
Comment (CelineBoudier) [reply aimmo-game/tests/functional/test_effect_expiry.py]: Yes, but shouldn't that be parametrised somewhere too?

File: aimmo-game/tests/functional/test_effect_expiry.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+        self.assertEqual(len(self.avatar.effects), 0)
+        self.assertEqual(self.avatar.attack_strength, 1)
+
+    def test_multiple_invulnerability_boost_pickup_expiry(self):
+        """
+        Ensure that each invulnerability boost effect expires at the appropriate time, even if the user stacks up on
+        them. One pickup will be at (1,0) and another at (3,0).
+        """
+        cell_two = self.game.game_state.world_map.get_cell(Location(3, 0))
```
Comment (CelineBoudier) [inline aimmo-game/tests/functional/test_effect_expiry.py]: a bit awkward as the other one is defined in the setup?

**Pull request merged**